### PR TITLE
OTG Test - RT-1.24: BGP 2-Byte and 4-Byte ASN support with policy

### LINKED ISSUE #2046: OTG: RT-1.24: BGP 2-Byte and 4-Byte ASN support with policy

### PR DESCRIPTION
**Pull request by octpetre**
The otg version of this test...which in found as RT-1.20 in the test plan.
Resolves #2046

### CODE REVIEW

File: feature/experimental/bgp/otg_tests/bgp_2byte_4byte_asn_policy_test/bgp_2byte_4byte_asn_policy_test.go [hunk 1]
```diff
@@ -0,0 +1,561 @@
...
+// juniperCLI returns Juniper CLI config statement.
+func juniperCLI() string {
+	return fmt.Sprintf(`
+	policy-options {
+		policy-statement %s {
+			term term1 {
+				from as-path match-as-path;
+				then reject;
+			}
+term term2 {
```
Comment (jasdeep-hundal) [inline feature/experimental/bgp/otg_tests/bgp_2byte_4byte_asn_policy_test/bgp_2byte_4byte_asn_policy_test.go:230]: Fix the indentation.

File: feature/experimental/bgp/otg_tests/bgp_2byte_4byte_asn_policy_test/bgp_2byte_4byte_asn_policy_test.go [hunk 2]
```diff
@@ -0,0 +1,561 @@
...
+		srcBgpPeer.SetPeerAddress(ateParams.peerIP).SetAsNumber(ateParams.localAS)
+		if connectionType == connInternal {
+			srcBgpPeer.SetAsType(gosnappi.BgpV4PeerAsType.IBGP)
+		} else {
+			srcBgpPeer.SetAsType(gosnappi.BgpV4PeerAsType.EBGP)
+		}
+		if asWidth == 2 {
+			srcBgpPeer.SetAsNumberWidth(gosnappi.BgpV4PeerAsNumberWidth.TWO)
+		}
+		prefixArr1 := strings.Split(prefixV4[0], "/")
```
Comment (jasdeep-hundal) [inline feature/experimental/bgp/otg_tests/bgp_2byte_4byte_asn_policy_test/bgp_2byte_4byte_asn_policy_test.go:418]: I think it's worth adding a function here (maybe titled `prefixAndLen`) that returns both the IP string and the length already casted as a uint32.

One nit: I think 'length' is more accurate than 'mask' here.

**Review by jasdeep-hundal — CHANGES_REQUESTED** [review]
@octpetre : Needs metadata sync and a couple small changes.

File: feature/experimental/bgp/otg_tests/bgp_2byte_4byte_asn_policy_test/bgp_2byte_4byte_asn_policy_test.go (hunk 1 above)
Comment (octpetre) [reply feature/experimental/bgp/otg_tests/bgp_2byte_4byte_asn_policy_test/bgp_2byte_4byte_asn_policy_test.go:230]: done

File: feature/experimental/bgp/otg_tests/bgp_2byte_4byte_asn_policy_test/bgp_2byte_4byte_asn_policy_test.go (hunk 2 above)
Comment (octpetre) [reply feature/experimental/bgp/otg_tests/bgp_2byte_4byte_asn_policy_test/bgp_2byte_4byte_asn_policy_test.go:418]: prefixAndLen added

File: feature/experimental/bgp/otg_tests/bgp_2byte_4byte_asn_policy_test/bgp_2byte_4byte_asn_policy_test.go [hunk 3]
```diff
@@ -0,0 +1,565 @@
...
+		network1.Addresses().Add().SetAddress(subnetAddr1).SetPrefix(subnetLen1).SetCount(1)
+		network1.AsPath().Segments().Add().SetAsNumbers([]uint32{55000, 4400, 3300})
+
+		network2 := srcBgpPeer.V4Routes().Add().SetName("bgpNeti2")
+		network2.SetNextHopIpv4Address(ateSrc.IPv4).
+			SetNextHopAddressType(gosnappi.BgpV4RouteRangeNextHopAddressType.IPV4).
+			SetNextHopMode(gosnappi.BgpV4RouteRangeNextHopMode.MANUAL)
+		network2.Addresses().Add().SetAddress(subnetAddr2).SetPrefix(subnetLen2).SetCount(1)
+		network2.AsPath().Segments().Add().SetAsNumbers([]uint32{55000, 7700})
+		network2.Communities().Add().SetAsNumber(200).SetAsCustom(1)
```
Comment (jasdeep-hundal) [inline feature/experimental/bgp/otg_tests/bgp_2byte_4byte_asn_policy_test/bgp_2byte_4byte_asn_policy_test.go:435]: @octpetre : Looks like we need to add SetType(gosnappi.BgpCommunityType.MANUAL_AS_NUMBER) to make that explicit. (I'm not sure what OTG is setting as a default here, but the test works after that change.)

File: feature/experimental/bgp/otg_tests/bgp_2byte_4byte_asn_policy_test/bgp_2byte_4byte_asn_policy_test.go [hunk 4]
```diff
@@ -0,0 +1,565 @@
...
+			SetNextHopMode(gosnappi.BgpV6RouteRangeNextHopMode.MANUAL)
+		network2.Addresses().Add().SetAddress(prefixArr2[0]).SetPrefix(uint32(mask2)).SetCount(1)
+		network2.AsPath().Segments().Add().SetAsNumbers([]uint32{55000, 7700})
+		network2.Communities().Add().SetAsNumber(200).SetAsCustom(1)
+
+		network3 := srcBgpPeer.V6Routes().Add().SetName("bgpNeti3")
+		network3.SetNextHopIpv6Address(ateSrc.IPv6).
+			SetNextHopAddressType(gosnappi.BgpV6RouteRangeNextHopAddressType.IPV6).
+			SetNextHopMode(gosnappi.BgpV6RouteRangeNextHopMode.MANUAL)
+		network3.Addresses().Add().SetAddress(prefixArr3[0]).SetPrefix(uint32(mask3)).SetCount(1)
```
Comment (jasdeep-hundal) [inline feature/experimental/bgp/otg_tests/bgp_2byte_4byte_asn_policy_test/bgp_2byte_4byte_asn_policy_test.go:480]: Same comment as above.

**Review by jasdeep-hundal — CHANGES_REQUESTED** [review]

File: feature/experimental/bgp/otg_tests/bgp_2byte_4byte_asn_policy_test/bgp_2byte_4byte_asn_policy_test.go (hunk 3 above)
Comment (octpetre) [reply feature/experimental/bgp/otg_tests/bgp_2byte_4byte_asn_policy_test/bgp_2byte_4byte_asn_policy_test.go:435]: Yes, the model doesn't define a default value...so I'm adding the type

File: feature/experimental/bgp/otg_tests/bgp_2byte_4byte_asn_policy_test/bgp_2byte_4byte_asn_policy_test.go (hunk 4 above)
Comment (octpetre) [reply feature/experimental/bgp/otg_tests/bgp_2byte_4byte_asn_policy_test/bgp_2byte_4byte_asn_policy_test.go:480]: Added SetType(gosnappi.BgpCommunityType.MANUAL_AS_NUMBER) to network2.Communities..

**Review by jasdeep-hundal — APPROVED** [review]

**Pull request merged**